compiler/frontend-spark/src/main/java/edu/snu/onyx/compiler/frontend/spark/JavaRDD.java

```diff
@@ -0,0 +1,196 @@
+import edu.snu.onyx.compiler.frontend.spark.transform.SerializableFunction;
+import org.apache.spark.serializer.KryoSerializer;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Stack;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Java RDD.
+ * @param <T> type of the final element.
+ */
+public final class JavaRDD<T extends Serializable> {
+  private final SparkContext sparkContext;
+  private final Integer parallelism;
+  private final Stack<LoopVertex> loopVertexStack;
+  private final DAG<IRVertex, IREdge> dag;
+  @Nullable private final IRVertex lastVertex;
+  private final KryoSerializer kryoSerializer;
+
+  /**
+   * Static method to create a JavaRDD object.
+   * @param sparkContext spark context containing configurations.
+   * @param parallelism parallelism information.
+   * @param <T> type of the resulting object.
+   * @return the new JavaRDD object.
+   */
+  static <T extends Serializable> JavaRDD<T> of(final SparkContext sparkContext, final Integer parallelism) {
+    return new JavaRDD<>(sparkContext, parallelism,
+        new DAGBuilder<IRVertex, IREdge>().buildWithoutSourceSinkCheck(), null);
+  }
+
+  /**
+   * Constructor.
+   * @param sparkContext spark context containing configurations.
+   * @param parallelism parallelism information.
+   * @param dag the current DAG.
+   * @param lastVertex last vertex added to the builder.
+   */
+  private JavaRDD(final SparkContext sparkContext, final Integer parallelism,
+                  final DAG<IRVertex, IREdge> dag, @Nullable final IRVertex lastVertex) {
+    this.loopVertexStack = new Stack<>();
+    this.sparkContext = sparkContext;
+    this.parallelism = parallelism;
+    this.dag = dag;
+    this.lastVertex = lastVertex;
+    this.kryoSerializer = new KryoSerializer(sparkContext.conf());
+  }
+
+  /////////////// TRANSFORMATIONS ///////////////
+
+  /**
+   * Set initialized source.
+   * @param initialData initial data.
+   * @return the Java RDD with the initialized source vertex.
+   */
+  JavaRDD<T> setSource(final Iterable<T> initialData) {
+    final DAGBuilder<IRVertex, IREdge> builder = new DAGBuilder<>(dag);
+
+    final IRVertex initializedSourceVertex = new InitializedSourceVertex<>(initialData);
+    initializedSourceVertex.setProperty(ParallelismProperty.of(parallelism));
+    builder.addVertex(initializedSourceVertex, loopVertexStack);
+
+    return new JavaRDD<>(this.sparkContext, this.parallelism,
+        builder.buildWithoutSourceSinkCheck(), initializedSourceVertex);
+  }
+
+  /**
+   * Map transform.
+   * @param func function to apply.
+   * @param <O> output type.
+   * @return the JavaRDD with the DAG.
+   */
+  public <O extends Serializable> JavaRDD<O> map(final SerializableFunction<T, O> func) {
```

Comment:
Ditto on supporting existing Spark programs.

Reply:
It's been extended.